### PR TITLE
fix master build

### LINKED ISSUE #42: Possible error in JointCommandHandle::set_cmd(double cmd)
https://github.com/ros-controls/ros2_control/blob/719cb6ca5d15d77e0a765546545a5807771697a2/hardware_interface/src/joint_command_handle.cpp#L53

I believe `THROW_ON_NULLPTR` should check `cmd_` and not `cmd`.

### PR DESCRIPTION
**Pull request by Karsten1987**
fixes #42 and replaces #39  as the master branch is not synced to `dashing` by `dashing_update` before being merged.

Signed-off-by: Karsten Knese <karsten@openrobotics.org>

### CODE REVIEW

**Comment by Karsten1987** [comment]
I can't really understand what's happening with the OSX actions. I can't reproduce the failure locally:

```
  2: [ RUN      ] TestControllerManager.update
  2: unknown file: Failure
  2: C++ exception with description "failed to initialized rcl init options: failed to find shared library of rmw implementation. Searched rmw_fastrtps_cpp, at /Users/runner/runners/2.164.0/work/ros2_control/ros2_control/ros_ws/src/ros2/rmw_implementation/rmw_implementation/src/functions.cpp:61, at /Users/runner/runners/2.164.0/work/ros2_control/ros2_control/ros_ws/src/ros2/rcl/rcl/src/rcl/init_options.c:55" thrown in SetUp().
  2: [  FAILED  ] TestControllerManager.update (0 ms)
```

@piraka9011 can you check internally what's happening here?

**Comment by Karsten1987** [comment]
first round of CI - using ros2 dashing:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9231/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4920/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7538/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9151/)
* Windows-container [Build Status](http://ci.ros2.org/job/ci_windows-container/33/)

**Comment by piraka9011** [comment]
> @piraka9011 can you check internally what's happening here? It feels to me as if either something wasn't sourced correctly or not all related packages were built.

Not sure why this is happening on Mac OS only. One thing I noticed is that you don't override your `SetUp` function (as suggested in the [docs](https://github.com/google/googletest/blob/master/googletest/docs/primer.md#test-fixtures-using-the-same-data-configuration-for-multiple-tests-same-data-multiple-tests)).

**Comment by Karsten1987** [comment]
addressing windows filesystem deprecation:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9232/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4921/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7538/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9152/)
* Windows-container [Build Status](http://ci.ros2.org/job/ci_windows-container/34/)

**Comment by Karsten1987** [comment]
next round:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9235/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4924/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7542/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9155/)
* Windows-container [Build Status](http://ci.ros2.org/job/ci_windows-container/37/)

**Comment by Karsten1987** [comment]
> Not sure why this is happening on Mac OS only. One thing I noticed is that you don't override your SetUp function (as suggested in the docs).

Looks like there is a discrepancy between the ROS2 buildfarm and the github actions as the OSX CI came back green.

**Comment by Karsten1987** [comment]
it's stays exciting:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9236/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4925/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7542/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9156/)
* Windows-container [Build Status](http://ci.ros2.org/job/ci_windows-container/38/)

File: ros2_control/package.xml [hunk 1]
```diff
@@ -16,6 +16,7 @@
   <exec_depend>controller_manager</exec_depend>
   <exec_depend>controller_parameter_server</exec_depend>
   <exec_depend>hardware_interface</exec_depend>
+  <exec_depend>test_robot_hardware</exec_depend>
```
Comment (bmagyar) [inline ros2_control/package.xml]: I'd opt for not adding testing packages to the metapackage as it doesn't form the main part of what we'd like someone to install when they install `ros2_control`. It's good to keep the installed bundle at a reasonable size.

**Comment by bmagyar** [comment]
@piraka9011 could you please double check the macos build again? This PR should have the changes from #39 but it still fails.

File: ros2_control/package.xml (hunk 1 above)
Comment (Karsten1987) [reply ros2_control/package.xml]: we can add them as a test dependency, but I generally don't agree with not listing them. If we build up to `ros2_control` all test packages should be compiled as well.

**Comment by Karsten1987** [comment]
> This PR should have the changes from #39 but it still fails.

Just to be clear. The PR is not failing as the ROS2 buildfarm returns green on all platforms. It's solely the GitHub action which might have a hiccup when not using the latest master, but building against dashing.

**Comment by brawner** [comment]
Unrelated to the failing github action, but the MSBuild warnings on ci_windows-container was fixed in https://github.com/ros2/rclcpp/pull/963, but probably won't be back ported.

**Comment by piraka9011** [comment]
@Karsten1987 @bmagyar I'm still looking into this, not sure why the GH action is failing.
Since the CI passes, I propose to move forward with this PR and I'll open a ticket to resolve this in `action-ros-ci`.

File: ros2_control/package.xml (hunk 1 above)
Comment (bmagyar) [reply ros2_control/package.xml]: Thanks

**Comment by bmagyar** [comment]
@piraka9011 for further reference, please post the issue you open here, or link this issue over there

**Pull request merged**